app/scripts/controllers/metametrics.js

```diff
@@ -630,11 +630,15 @@ export default class MetaMetricsController {
    * @returns {[]}
    */
   _getAllNFTsFlattened = memoize((allCollectibles = {}) => {
-    return Object.values(allCollectibles)
-      .reduce((result, chainNFTs) => {
-        return result.concat(Object.values(chainNFTs));
-      }, [])
-      .flat();
+    return Object.values(allCollectibles).reduce((result, chainNFTs) => {
+      let updatedResult;
+
+      Object.values(chainNFTs).forEach((addressNFTs) => {
```

Comment:
I don't actually think we need the `forEach` here.  I think we can simply do:
```
return result.concat(Object.values(chainNFTs))
```

Reply:
hm... I don't think this works

It might help to see this mock data I created:
```
const allCollectibles = {
          '0xac706cE8A9BF27Afecf080fB298d0ee13cfb9123': {
            56: [
              {
                address: '0xd2cea331e5f5d8ee9fb1055c297795937645de91',
                tokenId: '100',
              },
              {
                address: '0xd2cea331e5f5d8ee9fb1055c297795937645de91',
                tokenId: '101',
              },
              {
                address: '0x7488d2ce5deb26db021285b50b661d655eb3d3d9',
                tokenId: '99',
              },
            ],
          },
          '0xe04AB39684A24D8D4124b114F3bd6FBEB779c345: {
            69: [
              {
                address: '0x63d646bc7380562376d5de205123a57b1718184d',
                tokenId: '14',
              },
            ],
          },
};
```
 
replacing the logic with `return result.concat(Object.values(chainNFTs))` will produce this result
   
<img width="528" alt="Screenshot 2022-06-15 at 12 57 53 PM" src="https://user-images.githubusercontent.com/20778143/173894082-5101ee2a-8d69-484b-863e-34e94f04c52e.png">

Reply:
I was wrong -- I meant to say `return result.concat(…Object.values(chainNFTs))`

Reply:
You're totally right! I thought I was overengineering this when I wrote this out. Thank you @darkwing 🙏

Reply:
<img width="420" alt="Screenshot 2022-06-15 at 12 15 20 PM" src="https://user-images.githubusercontent.com/20778143/173896535-af49b18e-6017-4bea-b929-d4ded025b06d.jpeg">